horovod/common/operations.cc

```diff
@@ -1149,6 +1176,10 @@ void BackgroundThreadLoop(HorovodGlobalState& state, MPIContext& ctx) {
     cb(SHUT_DOWN_ERROR);
   }
 
+#if COMPILE_WITH_GLOO
```

Comment:
See comment above.  We should check against the environment variable that tells us if Gloo is being used at runtime here too.

Reply:
See above.